sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1826,4 +1826,13 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
     val df = spark.read.option("enforceSchema", false).csv(input)
     checkAnswer(df, Row("1", "2"))
   }
+
+  test("using the backward slash as the delimiter") {
+    val input = Seq("""abc\1""").toDS()
```

Comment:
If a user specified `\` as a delimiter, we should issue an exception message and let users add the escape symbol i.e., `\\`. It is weird to see both `\\` and `\` are representing the same thing `\`. We should be consistent for handling backslash in all the cases.

Reply:
I prohibited single backslash and throw an exception with a tip of using double backslash.